static/src/javascripts/projects/facia/modules/ui/container-show-more.js

```diff
@@ -47,32 +60,105 @@ define([
         var prefs = userPrefs.get(prefName, {
             type: 'session'
         });
-        return (prefs && prefs[containerId]) ? 'displayed' : 'hidden';
+        return (prefs && prefs[containerId]) ? STATE_DISPLAYED : STATE_HIDDEN;
     }
 
-    function showMore($container, button) {
+    function showMore(button) {
         fastdom.write(function () {
             /**
              * Do not remove: it should retain context for the click stream module, which recurses upwards through
              * DOM nodes.
              */
-            $container.toggleClass(className, button.state === 'displayed');
-            button.state = (button.state === 'hidden') ? 'displayed' : 'hidden';
-            setButtonState(button, button.state);
+            setButtonState(button, (button.state === STATE_HIDDEN) ? STATE_DISPLAYED : STATE_HIDDEN);
             updatePref(button.id, button.state);
         });
     }
 
-    function renderToDom($container, button) {
+    function renderToDom(button) {
         fastdom.write(function () {
-            $container.addClass(className)
+            button.$container.addClass(className)
                 .removeClass('js-container--fc-show-more')
-                .toggleClass(className, button.state === 'hidden');
+                .toggleClass(className, button.state === STATE_HIDDEN);
             // Initialise state, as it might be different from what was rendered server side based on localstorage prefs
             setButtonState(button, button.state);
         });
     }
 
+    function loadShowMore(pageId, containerId) {
+        return ajax({
+            url: '/' + pageId + '/show-more/' + containerId + '.json',
+            crossOrigin: true
+        });
```

Comment:
I now prefer we use @rich-nguyen's new ['ajax-promise`](https://github.com/guardian/frontend/blob/master/static/src/javascripts/projects/common/utils/ajax-promise.js) which returns a native promise rather than reqwest's fake one. This means you can also use a proper .then().catch() statement below.